scriptsOpenMC2GF/base_OpenMC2GF_MGXS.py

```diff
@@ -332,183 +300,126 @@ def CSV_to_GF():
+    beta_eff = pd.read_csv(f"{base_filepath_csvs}betaone.csv")['mean']
 
     # Lambda
-    df = pd.read_csv(f"{base_filepath_csvs}lambdaone.csv")
-    i=0
-    for row in df['mean']:
-        decay_const[i] = row
-        i += 1 
+    decay_const = pd.read_csv(f"{base_filepath_csvs}lambdaone.csv")['mean']
 
-    # Start writing nuclearData
-    f.write("\nzones \n ( \n")
+    # Start writing nuclearData with zones
+    fileString="\nzones \n ( \n"
 
     for zone in (OF_NAME):
-        f.write("\n%s \n{ \n" % zone)
-
-        f.write(" fuelFraction %s ; \n" % "{:.6e}".format(fuel_fraction) + "\n")
-
-        f.write(" IV nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in inv_velocity])) + "  );\n")
-
-        # Print groupwise diffusion coefficients
-        f.write("\n D nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in diffCoeff])) + "  );\n")
-
-        # Print groupwise nu sigma f
-        f.write("\n nuSigmaEff nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in nusigmaf])) + "  );\n")
-
-        # Print groupwise fission powers
-        f.write("\n sigmaPow nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in sigmaFissPow])) + "  );\n")
-
-        # Print scattering production matrix - P0 only
-        f.write("\n scatteringMatrixP0 %i %i (\n" % (energyGroups, energyGroups))
-
-        for i in range (energyGroups):
-            f.write(" (")
-            f.write(str(' '.join(['{:.6e}'.format(x) for x in scattering_P0[i][:]])) + " )\n")
-        f.write(" );\n")
-
-        # Print sigma absorption
-
-        f.write("\n sigmaDisapp nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in sigmaDisapp])) + "  );\n") # sigmaAbs 0.85 keff
-
-        f.write("\n chiPrompt nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in chi_prompt])) + "  );\n")
-
-        f.write("\n chiDelayed nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in chi_delayed])) + "  );\n")
-
-        # Print beta 
-        f.write("\n Beta nonuniform List<scalar> %i (" % precGroups + str(' '.join(['{:.6e}'.format(x) for x in beta_eff])) + "  );\n")
-
-        # Print decay constants for precursors
-        f.write("\n lambda nonuniform List<scalar> %i (" % precGroups + str(' '.join(['{:.6e}'.format(x) for x in decay_const])) + "  );\n")
-
+        fileString+="\n%s \n{ \n" % zone
+
+        fileString+=" fuelFraction %s ; \n" % "{:.6e}".format(fuel_fraction) + "\n"
+        fileString+=" IV nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in inv_velocity])) + "  );\n"
+
+        # Groupwise diffusion coefficientsm nuSigma and fission powers
+        fileString+="\n D nonuniform List<scalar> %i (" % energyGroups + str(' '.join(['{:.6e}'.format(x) for x in (diffCoeff)])) + "  );\n"

```

Comment:
One option (unsure if this actually works)
```suggestion
        fileString += f"\n D nonuniform List<scalar> {energyGroups} ({x:.6e for x in (diffCoeff)});\n"
```
Alternatively,
```suggestion
    fileString+=(
        "\n D nonuniform List<scalar> %i (" % energyGroups
        + str(' '.join(['{:.6e}'.format(x) for x in (diffCoeff)])) + " );\n"
    )
```